### PR TITLE
fix torch.distributed.rpc example incorrect usage

### LINKED ISSUE #112366: torch.distributed.rpc example errors
### 📚 The doc issue

https://pytorch.org/docs/stable/rpc.html#rpc

Usage of @torch.jit.script may not work with the current version of PyTorch and may result in type mismatch exceptions.

### Suggest a potential alternative/fix

yeah

### PR DESCRIPTION
**Pull request by littsk**
Fixes #112366

### CODE REVIEW

File: torch/distributed/rpc/api.py [hunk 1]
```diff
@@ -612,7 +612,7 @@ def remote(to, func, args=None, kwargs=None, timeout=UNSET_RPC_TIMEOUT):
 
         >>> # On both workers:
         >>> @torch.jit.script
-        >>> def my_script_add(t1, t2):
+        >>> def my_script_add(t1: torch.Tensor, t2: int):
```
Comment (H-Huang) [inline torch/distributed/rpc/api.py:615]: changing `t2`->`num` as @awgu suggested sounds good.

Comment (littsk) [reply torch/distributed/rpc/api.py]: Thanks for the suggestion, @H-Huang. I was also considering renaming the variables to tensor and scalar for even clearer context. Do you think that might be a more descriptive approach? Would love to hear your thoughts.

Comment (awgu) [reply torch/distributed/rpc/api.py]: Naming them tensor and scalar sounds good to me.

**Review by H-Huang — APPROVED** [review]
Thanks for the fix!

**Pull request closed without merging**